backend/src/handlers/libraries_handler.go

```diff
@@ -55,9 +58,43 @@ func (srv *Server) handleToggleLibraryVisibility(w http.ResponseWriter, r *http.
 	if err != nil {
 		return newInvalidIdServiceError(err, "library id")
 	}
-	if err := srv.Db.ToggleLibraryVisibility(id); err != nil {
+	library, err := srv.Db.ToggleLibraryVisibility(id)
+	if err != nil {
 		log.add("library_id", id)
 		return newDatabaseServiceError(err)
 	}
+	if srv.buckets != nil { //make sure to update value in bucket if exists
+		libraryBucket := srv.buckets[LibraryPaths]
+		updateLibraryBucket(libraryBucket, r.PathValue("id"), library, log)
+	}
 	return writeJsonResponse(w, http.StatusOK, "Library visibility updated successfully")
 }
+
+func updateLibraryBucket(libraryBucket nats.KeyValue, key string, library models.Library, log sLog) {
```

Comment:
if this is a method: we can avoid passing keyValue, and another pass library by ref

Reply:
made both changes